tests/conftest.py

```diff
@@ -129,8 +131,16 @@ def generate(code=False, **extra_params):
             "username": "n/a",
         }
         if code:
-            artifact["filename"] = "code.py"
-            artifact["language"] = "python"
+            if dynamics_only:
+                with open(f"{context_dir}/dynamics.json", "r") as f:
+                    dynamics = json.load(f)
+            else:
+                dynamics = {}
+            code_file = {
+                "dynamics": dynamics,
+                "language": "python",
+            }
+            artifact["files"] = {"code.py": code_file}
```

Comment:
This is a pretty significant change in the structure of the artifact object.
It removes `filename` and `language` adds `files` which is a dict (compared to `file_names` which is a list) where the dict values now have 'dynamics' and 'language'. Feels kind of ripe for confusion/problems.

Reply:
@mattprintz yeah this is a major change--we'd discussed with @YohannParis and @tangopapatime but I still think has big ramifications and is one of the major reasons for our TA1 open house Friday